src/features/page_view_event/aggregate/index.js

```diff
@@ -107,24 +115,30 @@ export class Aggregate extends AggregateBase {
     queryParameters.fp = firstPaint.current.value
     queryParameters.fcp = firstContentfulPaint.current.value
 
+    this.rumStartTime = now()
     const timeKeeper = this.agentRef.runtime.timeKeeper
     if (timeKeeper?.ready) {
-      queryParameters.timestamp = Math.floor(timeKeeper.correctRelativeTimestamp(now()))
+      queryParameters.timestamp = Math.floor(timeKeeper.correctRelativeTimestamp(this.rumStartTime))
+    } else {
+      queryParameters.rst = this.rumStartTime // we set this here in the feature instead of in the harvester to allow for a future retry to use the original RST
```

Comment:
I think you should report the rst as is at the time of retry. It's an accurate portrayal of what happened for that client because all the other events will follow this. You don't want a big time gap between PageView & rest.